sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -69,7 +72,16 @@ case class Exchange(newPartitioning: Partitioning, child: SparkPlan) extends Una
           }
         }
         val part = new HashPartitioner(numPartitions)
-        val shuffled = new ShuffledRDD[Row, Row, Row](rdd, part)
+        val shuffled = sort match {
+          case false => new ShuffledRDD[Row, Row, Row](rdd, part)
+          case true =>
+            val sortingExpressions = expressions.zipWithIndex.map {
+              case (exp, index) =>
+                new SortOrder(BoundReference(index, exp.dataType, exp.nullable), Ascending)
```

Comment:
This is not correct.  You don't want to construct your own `BoundReference`s otherwise the expression `a,b,c` will sort the same a `c,b,a`.